src/main/java/io/numaproj/numaflow/sinker/Service.java

```diff
@@ -18,9 +18,12 @@
 
 @Slf4j
 class Service extends SinkGrpc.SinkImplBase {
+    // sinkTaskExecutor is the executor for the sinker. It is a fixed size thread pool
+    // with the number of threads equal to the number of cores on the machine times 2.
     private final ExecutorService sinkTaskExecutor = Executors
             .newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 2);
 
+    // SHUTDOWN_TIME is the time to wait for the sinker to shut down, in seconds.
```

Comment:
Sorry, when I mentioned the magic number I meant it's not clear why the particular number was chosen. e.g. why we choose 30 seconds instead of 10 seconds. Adding an explicit comment on it will help other developers understand the full context and feel comfortable changing it.